buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -71,6 +72,8 @@ enum SizeClass {
     // TODO: Test if adding padding helps under contention
     //private long pad0, pad1, pad2, pad3, pad4, pad5, pad6, pad7;
 
+    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
nit: I wonder if we want to go with a `ReadWriteLock` as we also have quite some access that is read-only. That said maybe its something to consider as a followup.

Reply:
good point - racing good comment on https://github.com/netty/netty/pull/12622#issuecomment-1188747746